### PR TITLE
fix: partition of players is not updated

### PR DESCRIPTION
**Pull request by krzysztof-murczynski**
# Pull Request Description
Fix [#2737](https://github.com/decentraland/unity-explorer/issues/2737)

## What does this PR change?
Partition of avatars were not updated when avatar moved. I have introduced a new component that checks for if transform is dirty.

### Test Steps
1. First step
2. Second step
3. Expected result after step 2
4. ...

### Additional Testing Notes
- Note any edge cases to verify
- Mention specific areas that need careful testing
- List known limitations or potential issues

## Quality Checklist
- [ ] Changes have been tested locally
- [ ] Documentation has been updated (if required)
- [ ] Performance impact has been considered
- [ ] For SDK features: Test scene is included

## Code Review Reference
Please review our [Code Review Standards](https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md) before submitting.

### CODE REVIEW

File: Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs [hunk 1]
```diff
@@ -23,6 +31,41 @@ public CharacterTransform(Transform transform)
...
+            IsDirty = false;
+        }
+        
+        private void TrySetDirty(Vector3 newPosition)
+        {
+            if (IsDirty) return;
+            
+            float distance = CheapDistance(oldPosition, newPosition);
+
+            if (!(distance >= MINIMAL_DISTANCE_DIFFERENCE)) return;
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs:57]: You could invert the comparison for clarity to remove negation: it's simply `<`, right?

File: Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs [hunk 2]
```diff
@@ -8,13 +8,21 @@ namespace DCL.Character.Components
     ///     Character Transform is not a regular <see cref="TransformComponent" />
     ///     as it's driven by physics and does not have a parent and children
     /// </summary>
-    public readonly struct CharacterTransform : IPoolableComponentProvider<Transform>
+    public struct CharacterTransform : IPoolableComponentProvider<Transform>
     {
+        private const float MINIMAL_DISTANCE_DIFFERENCE = 0.01f;
+        
         public readonly Transform Transform;
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs:15]: Have you checked if there are no other direct assignments of the position that can break `IsDirty` logic?

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterObject.Tests.asmref [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "reference": "GUID:7213a1d35667f9b40be6aca8cbfcee23"
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterObject.Tests.asmref:2]: You should use `EditModeTests` unless you require some very specific logic connected to the Physics, Player Loop or networking that doesn't function properly in the edit mode

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+            
+            // Assert
+            ref CharacterTransform dirtyAfter = ref globalWorld.Get<CharacterTransform>(dirtyEntity);
+            ref CharacterTransform cleanAfter = ref globalWorld.Get<CharacterTransform>(cleanEntity);
+            
+            Assert.IsFalse(dirtyAfter.IsDirty, "Dirty entity should be cleared");
+            Assert.IsFalse(cleanAfter.IsDirty, "Clean entity should remain clean");
+            
+            // Cleanup
+            Object.DestroyImmediate(cleanGameObject);
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs:223]: You shouldn't clean-up in the test itself: if `Assert` fails, the rest of the code won't be executed: you've done it correctly with `testGameObject`: consider sticking to the same principle, you can have a list of objects if needed

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+using ECS.TestSuite;
+using NSubstitute;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.Pool;
+using Utility.PriorityQueue;
+
+namespace DCL.Character.Tests
+{
+    public class CharacterTransformDirtyFlagShould : UnitySystemTestBase<PartitionGlobalAssetEntitiesSystem>
```
Comment (mikhail-dcl) [inline Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs:26]: I see you've done it very-very carefully 😄 I couldn't expect such level of attention for the test 👍

**Review by mikhail-dcl — CHANGES_REQUESTED** [review]
Looks very good and concise 💜 I simply have minor comments

File: Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs (hunk 1 above)
Comment (krzysztof-murczynski) [reply Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs:57]: Absolutely right, that's probably rider's shenanigans, which I have missed 😅

File: Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs (hunk 2 above)
Comment (krzysztof-murczynski) [reply Explorer/Assets/DCL/Character/CharacterObject/Components/CharacterTransform.cs:15]: @mikhail-dcl There are, interpolation is moving avatar as well, but my reasoning with that is that calculation it on input received should cover all of these cases without unnecessarily calculating it several times

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterObject.Tests.asmref (hunk 3 above)
Comment (krzysztof-murczynski) [reply Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterObject.Tests.asmref:2]: Oh okey, I checked and it works without any further changes, nice :D

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs (hunk 5 above)
Comment (krzysztof-murczynski) [reply Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs:26]: I did not expect being praised for this one 😅 Thank you! (well, I hope you didn't meant it sarcastically 😅)

File: Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs (hunk 4 above)
Comment (mikhail-dcl) [reply Explorer/Assets/DCL/Character/CharacterObject/Tests/CharacterTransformDirtyFlagShould.cs]: Thanks

**Review by DafGreco — APPROVED** [review]
✔️  PR reviewed and approved by QA on [ Platform ] following instructions playing both happy and un-happy path

Regressions for this ticket had been performed in order to verify that the normal flow is working as expected:

- [✔️  ] Backpack and wearables in world
- [✔️  ] Emotes in world and in backpack
- [ ✔️ ] Teleport with map/coordinates/Jump In
- [✔️  ] Chat and multiplayer
- [✔️  ] Profile card
- [✔️  ] Settings


Evidence :

https://github.com/user-attachments/assets/4ea8b890-d1ba-4731-a0e5-1be5ed6d8365


https://github.com/user-attachments/assets/a7240794-b118-47ac-ab95-18fd53411bbd
